src/commands/reminders/remind.ts

```diff
@@ -11,38 +11,53 @@ export default class extends Command {
 	description = `Have ${BOT.NAME} give you a reminder.`;
 	usage = '<reminder> | <duration> | [repeat]';
 	extendedHelp = 'Reminders can be set to repeat daily or weekly.';
-
-	run(msg: Message, [reminder]: [Reminder]): Promise<Message> {
-		msg.client.mongo.collection(DB.REMINDERS).insertOne(reminder);
-
-		return msg.channel.send(`I'll remind you about that at ${reminderTime(reminder)}.`);
-	}
-
-	argParser(msg: Message, input: string): [Reminder] {
-		const [content, rawDuration, rawRepeat] = input.split('|').map(part => part.trim());
-		const weekWords = ['w', 'week', 'weekly'];
-		const dayWords = ['d', 'day', 'daily'];
-
+	options: ApplicationCommandOptionData[] = [
+		{
+			name: 'content',
+			description: 'What you\'d like to be reminded of',
+			type: 'STRING',
+			required: true
+		},
+		{
+			name: 'duration',
+			description: 'When you\'d like to be reminded',
+			type: 'STRING',
+			required: true
+		},
+		{
+			name: 'repeat',
+			description: 'How often you want the reminder to repeat',
+			choices: [{ name: 'Daily', value: 'daily' }, { name: 'Weekly', value: 'weekly' }],
+			type: 'STRING',
+			required: false
```

Comment:
I recently made this change in a not yet public version of Steve but you should have the repeat just be another duration. It would also take some modification of the reminder piece and the view reminder command though.